lib/cmgd_bcknd_client.c

```diff
@@ -43,6 +45,64 @@ const char *cmgd_bcknd_client_names[CMGD_CLIENT_NAME_MAX_LEN] = {
 	"Unknown/Invalid",		/* CMGD_BCKND_CLIENT_ID_MAX */
 };
 
+DEFINE_MTYPE_STATIC(LIB, CMGD_BCKND_BATCH, "CMGD backend transaction batch data");
+DEFINE_MTYPE_STATIC(LIB, CMGD_BCKND_TRXN, "CMGD backend transaction data");
+DEFINE_MTYPE_STATIC(LIB, CMGD_BCKND_TRXN_REQ,"CMGD backend transaction request");
+DEFINE_MTYPE_STATIC(LIB, CMGD_BCKND_TRXN_SETCFG_REQ, "CMGD backend transaction set request");
+DEFINE_MTYPE_STATIC(LIB, CMGD_BCKND_TRXN_GETDATA_REQ, "CMGD backend transaction get request");
+
+const char *cmgd_bcknd_client_names[] = {CMGD_BCKND_CLIENT_STATICD, CMGD_BCKND_CLIENT_BGPD};
+
+typedef enum cmgd_bcknd_trxn_event_ {
+	CMGD_BCKND_TRXN_PROC_SETCFG = 1,
+	CMGD_BCKND_TRXN_PROC_GETCFG,
+	CMGD_BCKND_TRXN_PROC_GETDATA
+} cmgd_bcknd_trxn_event_t;
+
+typedef struct cmgd_bcknd_set_cfg_req_ {
+	struct nb_cfg_change cfg_changes[CMGD_MAX_CFG_CHANGES_IN_BATCH];
+	uint16_t num_cfg_changes;
+} cmgd_bcknd_set_cfg_req_t;
+
+typedef struct cmgd_bcknd_get_data_req_ {
+	char *xpaths[CMGD_MAX_NUM_DATA_IN_BATCH];
+	uint16_t num_xpaths;
+} cmgd_bcknd_get_data_req_t;
+
+PREDECL_LIST(cmgd_bcknd_trxn_req_list);
+typedef struct cmgd_bcknd_trxn_req_ {
+	cmgd_bcknd_trxn_event_t event;
+	union {
+		cmgd_bcknd_set_cfg_req_t *set_cfg;
+		cmgd_bcknd_get_data_req_t *get_data;
+	} req;
+
+	struct cmgd_bcknd_trxn_req_list_item list_linkage;
+} cmgd_bcknd_trxn_req_t;
+DECLARE_LIST(cmgd_bcknd_trxn_req_list, cmgd_bcknd_trxn_req_t, list_linkage);
+
+PREDECL_LIST(cmgd_bcknd_batch_list);
+typedef struct cmgd_bcknd_batch_ctxt_ {
+	//add private data
+	uint64_t batch_id;
+	struct cmgd_bcknd_trxn_req_list_head trxn_data;
```

Comment:
This doesn't look okay. A single batch cannot span over multiple transaction requests.

Reply:
Its the other way.